### PR TITLE
Update the webui to the new design

### LINKED ISSUE #2034: Update the WebUI's design
Update the WebUI interface based on the mockups by Caroline.

### Evaluate

- [x] Evaluate if and which Angular Material components could be used.

### Update

- [x] Update icons and colors where needed
- [x] Update License Page
- [x] Update Container Page
- [x] Update Tokens Page (#2036)
- [x] Update Channels Page (#2037)
- [x] Remove PrimeNG dependency

### LINKED ISSUE #2036: Update Tokens page to the new design
Update the tokens page to the new design based on the screens in the mockups.

- [x] Move Network events to a new page
- [x] Move Token events to a new page
- [x] Add actions to be able to navigate to the pages
- [x] Update Dialogs to Material
  - [x] Token Registration Dialog
  - [x] Join Token Network Dialog
  - [x] Transfer Tokens Dialog
  - [x] (Not currently used) Token Swap Dialog
- [x] Update events page to material

### LINKED ISSUE #2037: Update channels page to the new design
This is an issue to track work on the UI design update on the channels page.
- [x] Update Channels Page to the new design
- [x] Move channel events to a new Component with its own route
- [x] Update Dialogs
   - [x] Transfer Dialog
   - [x] Deposit Dialog

### LINKED ISSUE #2192: Show a notification on webUI when a transfer is received and new channel is opened with us
## Problem Definition

As when a outgoing transfer succeed, we should show a success notification when receiving a transfer on any of the open channels, or a new channel is opened with us. Maybe also add some _highlight_ animation to the channel and balance too, if currently on the channels screen.
Related: #2188

### LINKED ISSUE #2197: WebUI must handle token decimals
## Problem Definition

Right now, webUI shows tokens as Raiden sees it: only token `wei`s, as integers. The webUI/client is responsible for managing token decimals, and should handle it gracefully. Each token contract have a `decimals()` method which returns an integer for the amount of decimals this token is configured with. UI `input type="number"`s should convert from decimal number to integer before sending the values to the REST API (`raiden_amount = int(input_amount * 10**decimals)`) and prevent user from entering more decimals than token supports, and places where any amount/deposit/balance is displayed should take it into account (`ui_amount = raiden_amount / 10**decimals`). `BigNumber` may be required. `RaidenService` may do it before handling data to the rest of the components.

### PR DESCRIPTION
**Pull request by kelsos**
* Replaces font awesome/bootstrap/primeng with angular material/material icons/flex-layout (growl replaced with ngx-toastr).
* Updates npm dependencies .
* Removes old css rules, updates css.
* Updates all the existing component templates to use components from the angular material.
* Updates all dialogs to material dialogs.
* Moves event listing to new components/routes 
* Remove `RaidenService` references from any dialog, move the logic to the host component (except SwapDIalogComponent).
* `SharedService` now has different methods for each message type.
* Move error handling and message creation to `RaidenService`.
* Adds notification when the a new channel opens.
* Adds notification when balance increases for a channel.
* Adds handling of token decimals.

Closes #2034 
Closes #2036
Closes #2037 
Closes #2192
Closes #2197

### CODE REVIEW

**Comment by pcppcp** [comment]
idea for an improvement: what I miss as an user is a list of received/send payments, this would be really cool to have

**Comment by kelsos** [comment]
@pcppcp Maybe open an issue about it? The endpoint is available so it shouldn't be hard to implement

**Comment by alexhmyang** [comment]
https://raiden-network.readthedocs.io/en/stable/webui_tutorial.html 
how to install web ui, we cannot browse it from chrome, can you write more clearly ? thank you

File: raiden/ui/web/src/app/app.component.ts [hunk 1]
```diff
@@ -7,17 +10,42 @@ import { SharedService } from './services/shared.service';
...
-        this.raidenService.getRaidenAddress()
-            .subscribe((address) => this.raidenAddress = address);
+        this.raidenService.getRaidenAddress().subscribe((address) => this.raidenAddress = address);
+        this.sub = this.sharedService.pendingRequests.subscribe((pendingRequests) => {
+            setTimeout(() => {
+                this.pendingRequests = pendingRequests;
+            });
+        });
+
+        this.balanceCheckerService.startMonitoring();
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/app.component.ts]: I don't think you need these `startMonitoring`/`stopMonitoring` in the `balanceCheckerService`: a more _Observable-centric_ approach should be to expose a shared `Observable` there (maybe through a public readonly property or getter) directly, containing your logic directly inside the stream, and then subscribing here (in the top level component), possibly discarding any value, the point is just the subscription, so if the observable is subscribed, it acts as `startMonitoring`, and when last subscription is unsubscribed, it'll automatically stop monitoring. And, therefore, if it's just a more complex observable, you may as well just put it inside `RaidenService` directly (or keep it in a separate service too, if you want them separate)

File: raiden/ui/web/src/app/components/channel-events/channel-events.component.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+@Component({
+    selector: 'app-channel-events',
+    templateUrl: './channel-events.component.html',
+    styleUrls: ['./channel-events.component.css']
+})
+export class ChannelEventsComponent implements OnInit {
+
+    public eventsParam: Observable<EventsParam>;
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/components/channel-events/channel-events.component.ts]: Small nit-pick, but you could stick to the `$` suffix on observable variables

File: raiden/ui/web/src/app/app.component.ts [hunk 3]
```diff
@@ -7,17 +10,42 @@ import { SharedService } from './services/shared.service';
...
+        private balanceCheckerService: ChannelPollingService
+    ) {
+    }
 
     ngOnInit() {
-        this.raidenService.getRaidenAddress()
-            .subscribe((address) => this.raidenAddress = address);
+        this.raidenService.getRaidenAddress().subscribe((address) => this.raidenAddress = address);
+        this.sub = this.sharedService.pendingRequests.subscribe((pendingRequests) => {
+            setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (kelsos) [reply raiden/ui/web/src/app/app.component.ts]: Without it I was getting an `ExpressionChangedAfterItHasBeenCheckedError` spammed. After checking for ways to get it solved this was one of the suggested workarounds, and the only think of the suggested workarounds that actually stopped the spamming of the error.

[the remark this answers is not in the document]
Comment (andrevmatos) [reply raiden/ui/web/src/app/app.component.ts]: Right.. that warning can be safely ignored though, it just happens on dev builds, but got it =)

File: raiden/ui/web/src/app/app.component.ts (hunk 1 above)
Comment (kelsos) [reply raiden/ui/web/src/app/app.component.ts]: Ok I am on it :)

File: raiden/ui/web/src/app/components/channel-events/channel-events.component.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                const channelInformation: ChannelInformation = {
+                    token_address: params.get('token_address'),
+                    partner_address: params.get('partner_address')
+                };
+
+                return {
+                    channel: channelInformation
+                };
+            }),
+            tap((eventsParam: EventsParam) => console.log(eventsParam))
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/components/channel-events/channel-events.component.ts]: nit-pick: putting `console.log` directly as the callback for tap work as well =)

Comment (kelsos) [reply raiden/ui/web/src/app/components/channel-events/channel-events.component.ts]: Oh ok thanks :), but I am not sure even if there is a valid reason for it being there anyway so I might as well remove it completely.

File: raiden/ui/web/src/app/components/channel-events/channel-events.component.ts (hunk 2 above)
Comment (kelsos) [reply raiden/ui/web/src/app/components/channel-events/channel-events.component.ts]: I have to check if we can enforce it with some kind of lint rule.

**Pull request merged**